### PR TITLE
Re-org: Rename plugin/ files and classes

### PR DESCRIPTION
**Pull request by JohannesLorenz**
Re-org PR! (#5592)

**Description**

This fixes casing of everything in plugins/ that is currently lowercase but should be uppercase.
Additional fixups:

* style.css: Fix opl2 -> Opulenz
* bs/sr.ts: Fix outdated entries

**Review**

This PR changes only code, i.e. no library names or strings for savefiles/presets. There are changes in the locales and themes though, but I verified all of them using scripts.

* Code review: I checked all files, but it can't be harm to have someone cross-check it.
* Test review: Check that
  - most native plugins still work (maybe check songs with many different native plugins)
  - most native plugins still look correct (no style issues)
  - most tooltips still work (and are translated, if they were before)

**Open TODO** (but it can be done later)

Remove duplicate translations. The following warnings show up while compiling:

```
Warning: dropping duplicate messages in '/home/johannes/cprogs/lmms/master/data/locale/es.qm':

* Context: PatchesDialog
* Source: Patch

* Context: PatchesDialog
* Source: Program selector

* Context: PatchesDialog
* Source: Bank

* Context: PatchesDialog
* Source: Bank selector

* Context: PatchesDialog
* Source: Qsynth: Channel Preset

* Context: PatchesDialog
* Source: Cancel

* Context: PatchesDialog
* Source: OK

* Context: PatchesDialog
* Source: Name
```

### CODE REVIEW

File: plugins/Sf2Player/Sf2Player.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@
  *
  */
 
-#include "sf2_player.h"
+#include "Sf2Player.h"
```
[the remark this answers is not in the document]
Comment (JohannesLorenz) [reply plugins/Sf2Player/Sf2Player.cpp:26]: I do what @tresf said :laughing: And he is right IMO. If I look into cmake/modules/PluginList.cmake, we did use `Vst`, `Lv2`, `Midi` etc. The `SF2` in the class names is an error that I need to fix.

File: plugins/Sf2Player/PatchesDialog.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * patches_dialog.cpp - display sf2 patches
+ * PatchesDialog.cpp - display sf2 patches
```
[the remark this answers is not in the document]
Comment (JohannesLorenz) [reply plugins/Sf2Player/PatchesDialog.cpp:2]: Sounds good. Will do.

File: plugins/Lb302/Lb302.cpp [hunk 3]
```diff
@@ -1,14 +1,14 @@
 /*
- * lb302.cpp - implementation of class lb302 which is a bass synth attempting
+ * Lb302.cpp - implementation of class LB302 which is a bass synth attempting
```
[the remark this answers is not in the document]
Comment (JohannesLorenz) [reply plugins/Lb302/Lb302.cpp:2]: Lb302, after the mentioned discussion :grinning:

[the remark this answers is not in the document]
Comment (JohannesLorenz) [reply plugins/Lb302/Lb302.cpp:2]: Fixed, but only in those 2 file comments. All other occurrences seemed to be not the class name, but the instrument name, so LB302 seems correct in the other cases.

**Review by Veratil — APPROVED** [review]
LGTM!

**Comment by JohannesLorenz** [comment]
> LGTM!

Perfect. Thanks for the 2 reviews!

Now the PR is open for voluntary testing. I'll merge it on the weekend if no one volunteers.

File: cmake/modules/PluginList.cmake [hunk 4]
```diff
@@ -100,7 +100,7 @@ ENDIF()
 IF(MSVC)
 	SET(MSVC_INCOMPATIBLE_PLUGINS
 		LadspaEffect
-		zynaddsubfx
+		Zynaddsubfx
```
Comment (DomClark) [inline cmake/modules/PluginList.cmake:103]:
```suggestion
		ZynAddSubFx
```

File: .gitmodules [hunk 5]
```diff
@@ -5,7 +5,7 @@
 	path = src/3rdparty/rpmalloc/rpmalloc
 	url = https://github.com/mjansson/rpmalloc.git
 [submodule "plugins/zynaddsubfx/zynaddsubfx"]
-	path = plugins/zynaddsubfx/zynaddsubfx
+	path = plugins/ZynAddSubFx/zynaddsubfx
```
Comment (DomClark) [inline .gitmodules:8]: Should the case of these paths match?

Comment (Veratil) [reply .gitmodules:8]: They would matter on Linux/Mac, but I think on Windows it ignores case. Good catch!

You'll also wants to updated the `[submodule "plugins/ZynAddSubFx/zynaddsubfx"]` section header.

**Comment by JohannesLorenz** [comment]
@DomClark Thanks for the review. Good found x2. I found occurrences in even more files:

```
.gitignore
.gitmodules
cmake/modules/CheckSubmodules.cmake
cmake/modules/PluginList.cmake
debian/copyright
debian/patches/build-amd64-20181013.patch
doc/Doxyfile.in
```

Before I commit, I'll now check these files for other plugins and check the whole file tree for more files.

One thing where I'm not sure:

https://github.com/LMMS/lmms/blob/9628a2b512dba7086532bad2adee618bffa9fb53/CMakeLists.txt#L677-L686

Should I rename `plugins/zynaddsubfx` to `plugins/ZynAddSubFx` here, too? I don't know where these `extra/` files for this "win 32 build" come from. The last commit really dealing with it is eb2f5e9cc2 , though @jasp00 has also committed in that area (5b2283feec).

**Comment by JohannesLorenz** [comment]
I pushed the fixes, and also checked for all other files in the LMMS tree. For the strings that I changed in the patches, I verified all new locations with a script.

As discussed in Discord, I removed the CMakeLists.txt "package ZynAddSubFX into win32 build" section. @DomClark , please verify the Windows installer.

**Review by DomClark — APPROVED** [review]

**Review by Veratil — APPROVED** [review]

**Comment by JohannesLorenz** [comment]
Thanks a lot you two. I will merge tomorrow.

**Pull request merged**